tests/test_outcomes_api.py

```diff
@@ -152,3 +165,129 @@ def test_happy_path_querying(self):
+        self.generate_outcomes(
+            org_id=1,
+            project_id=1,
+            num_outcomes=1,
+            outcome=0,
+            category=DataCategory.ERROR,
+            time_since_base=timedelta(minutes=500),
+        )
+        self.generate_outcomes(
+            org_id=1,
+            project_id=1,
+            num_outcomes=1,
+            outcome=0,
+            category=None,  # should be counted as an Error
+            time_since_base=timedelta(minutes=500),
+        )
+        self.generate_outcomes(
+            org_id=1,
+            project_id=1,
+            num_outcomes=1,
+            outcome=0,
+            category=DataCategory.SECURITY,
+            time_since_base=timedelta(minutes=500),
+        )
+        self.generate_outcomes(
+            org_id=1,
+            project_id=1,
+            num_outcomes=1,
+            outcome=0,
+            category=DataCategory.TRANSACTION,
+            time_since_base=timedelta(minutes=500),
+        )
+        self.generate_outcomes(
+            org_id=1,
+            project_id=1,
+            num_outcomes=1,
+            outcome=0,
+            quantity=6,
+            category=DataCategory.SESSION,
+            time_since_base=timedelta(minutes=500),
+        )
+        self.generate_outcomes(
+            org_id=1,
+            project_id=1,
+            num_outcomes=1,
+            outcome=0,
+            quantity=4,
+            category=DataCategory.SESSION,
+            time_since_base=timedelta(minutes=500),
+        )
+        self.generate_outcomes(
+            org_id=1,
+            project_id=1,
+            num_outcomes=1,
+            outcome=0,
+            category=DataCategory.ATTACHMENT,
+            quantity=65536,
+            time_since_base=timedelta(minutes=500),
+        )
+        self.generate_outcomes(
+            org_id=1,
+            project_id=1,
+            num_outcomes=1,
+            outcome=0,
+            category=DataCategory.ATTACHMENT,
+            quantity=16384,
+            time_since_base=timedelta(minutes=500),
+        )
+
+        from_date = self.format_time(self.base_time + timedelta(minutes=400))
+        to_date = self.format_time(self.base_time + timedelta(minutes=600))
+        response = self.app.post(
+            self.endpoint,
+            data=json.dumps(
+                {
+                    "dataset": "outcomes",
+                    "aggregations": [
+                        ["sum", "times_seen", "times_seen"],
+                        ["sum", "quantity", "quantity_sum"],
+                    ],
+                    "from_date": from_date,
+                    "selected_columns": [],
+                    "to_date": to_date,
+                    "organization": 1,
+                    "conditions": [
+                        ["timestamp", ">", from_date],
+                        ["timestamp", "<=", to_date],
+                    ],
+                    "groupby": ["category"],
```

Comment:
Sentry is currently sending queries that don't include category (as it doesn't know that outcomes has categories) via TSDB.

Should the snuba API apply a condition of `category=0` if no category condition is provided as a way to provide a smooth transition and consistent results until we can update sentry with an additional condition inside of TSDB?

Adding conditions to sentry before this ships will be hard as snuba will raise errors on the category column missing, and having snuba ignore errors/drop columns seems like a more opaque interface than adding conditions.

Reply:
Why is it wrong for Sentry to send queries without category? If a query without category is sent, how will the results be different from what they are now?

Reply:
I will test out how TSDB handles queries on the sentry side and ensure that it still works as expected with the extra column.

I believe we'll have to change the current TSDB implementation to filter everything but aforementioned Error, Default, and Security in current queries. This change would be deployed after migrations have been ran but before the consumer change is deployed.

Reply:
confirmed that testing locally adding quantity and category do not affect current queries. Before the consumer changes are deployed, we can add this change https://github.com/getsentry/sentry/pull/23927 to our TSDB consumers (I need to test this some more, but will look something like that), so that current org stats page is not changed.

Reply:
> Why is it wrong for Sentry to send queries without category? If a query without category is sent, how will the results be different from what they are now?

Once snuba starts recording outcomes for more than errors, if sentry doesn't include a category condition, then it will get results for all categories and render them as errors. This would be misleading as the totals would include errors, transactions and attachment payloads that were received.

Reply:
So, just to confirm, the sequence would be this, right?
1) migration PR approved (James will look into this tomorrow morning)
2) migration PR merged
3) migraiton applied in prod
4) sentry change to add the category condition
5) this change is deployed to fill in the new fields.

Reply:
@fpacifici yup!

Reply:
getsentry/sentry#23927 is deployed, so this PR is next!